app/controllers/users/service_provider_inactive_controller.rb

```diff
@@ -0,0 +1,19 @@
+module Users
+  class ServiceProviderInactiveController < ApplicationController
+    include FullyAuthenticatable
+
+    def index
+      analytics.track_event(Analytics::SP_INACTIVE_VISIT)
+      @sp_name = I18n.t('service_providers.errors.generic_sp_name')
+
+      issuer = sp_session[:issuer]
+      if issuer.present?
+        service_provider = ServiceProvider.from_issuer(issuer)
+        @sp_name = service_provider.friendly_name || @sp_name
+      end
```

Comment:
Since this inherits from ApplicationController, we can simplify this whole thing to:

```suggestion
      @sp_name = sp_from_sp_session&.friendly_name || I18n.t('service_providers.errors.generic_sp_name')
```

Reply:
Ah! I didn't know about that. That's much nicer. Thanks!